### PR TITLE
Add MarkdownContentAttribute

### PR DESCRIPTION
**Pull request by vicancy**
And support *content

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/BaseModelAttributeHandler.cs [hunk 1]
```diff
@@ -73,6 +73,11 @@ public virtual void Handle(object obj, HandleModelAttributesContext context)
             }
         }
 
+        protected virtual bool ShouldHandle(PropInfo currentPropInfo, object declaringObject, HandleModelAttributesContext context)
+        {
+            return currentPropInfo.Attr != null;
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/BaseModelAttributeHandler.cs]: "urrentPropInfo?.Attr"?

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                if (context.EnableContentPlaceholder)
+                {
+                    var currentValue = currentPropInfo.Prop.GetValue(declaringObject) as string;
+                    if (currentValue != null && IsPlaceholderContent(currentValue))
+                    {
+                        return true;
+                    }
+                }
+
+                return currentPropInfo.Attr != null;
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs]: need this line call base.ShouldHandle() instead?

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            private string Markup(string content, HandleModelAttributesContext context)
+            {
+                if (string.IsNullOrEmpty(content))
+                {
+                    return content;
+                }
+
+                if (context.EnableContentPlaceholder)
+                {
+                    if (IsPlaceholderContent(content))
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs]: minor: this if may merge with the if above?

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+                var host = context.host;
+                var mr = host.Markup(content, context.FileAndType);
+                context.LinkToUids.AddRange(mr.LinkToUids);
+                AddRange(context.LinkToFiles, mr.LinkToFiles);
+                AddRange(context.FileLinkSources, mr.FileLinkSources);
+                AddRange(context.UidLinkSources, mr.UidLinkSources);
+                return mr.Html;
+            }
+
+            private static void AddRange(HashSet<string> left, IEnumerable<string> right)
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs]: can use HashSet.UnionWith()?

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    using System;
+    using System.Collections;
+    using System.Collections.Concurrent;
+    using System.Reflection;
+
+    using Microsoft.DocAsCode.DataContracts.Common.Attributes;
+    using System.Linq;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using Plugins;
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs]: group using

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/BaseModelAttributeHandler.cs (hunk 1 above)
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/BaseModelAttributeHandler.cs]: it is a protected method, and we know currentPropInfo can never be null

---
In reply to: [93932946](https://github.com/dotnet/docfx/pull/1064#discussion_r93932946) [](ancestors = 93932946)

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+    public class MarkdownContentHandler : IModelAttributeHandler
+    {
+        private readonly static ConcurrentDictionary<Type, MarkdownContentHandlerImpl> _cache = new ConcurrentDictionary<Type, MarkdownContentHandlerImpl>();
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs]: >private readonly static ConcurrentDictionary [](start = 8, length = 44)

placeholderContentAfterMarkup inside MarkdownContentHandlerImpl may conflict when paralell

Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs]: nice catch, Impl should not keep status

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Pull request closed without merging**